src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -89,17 +90,17 @@ private User userRegistrationDtoToUser(UserRegistrationDto userRegistrationDto)
 
     @Override
     public User loginUserWithOAuth2(DefaultOidcUser oauth2User) {
-        var optionalUser = findByUsernameOrEmail(oauth2User.getEmail());
-        optionalUser.filter(Predicate.not(this::isUserHasNoOAuthOrValidatedEmail))
-                .map(User::getEmail)
+        Optional<UserProjection> optionalUserProjection = userRepository.findUserProjectionByEmail(oauth2User.getEmail());
```

Comment:
You don't need to specify optional into the variable name once you have a predefined type